src/widgets/table.rs

```diff
@@ -308,80 +395,72 @@ impl<'a> Table<'a> {
         let widths = widths.into_iter().map(|c| *c.as_ref()).collect_vec();
         ensure_percentages_less_than_100(&widths);
         Self {
-            block: None,
-            style: Style::default(),
+            rows: rows.into_iter().collect(),
             widths,
             column_spacing: 1,
-            highlight_style: Style::default(),
-            highlight_symbol: None,
-            header: None,
-            rows: rows.into_iter().collect(),
-            highlight_spacing: HighlightSpacing::default(),
+            // Note: None is not the default value for SegmentSize, so we need to explicitly set it
             segment_size: SegmentSize::None,
+            ..Default::default()
         }
     }
 
-    /// Creates a custom block around a [`Table`] widget.
+    /// Set the rows
     ///
-    /// The `block` parameter is of type [`Block`]. This holds the specified block to be
-    /// created around the [`Table`]
+    /// The `rows` parameter accepts any value that can be converted into an iterator of [`Row`]s.
+    /// This includes arrays, slices, and [`Vec`]s.
+    ///
+    /// # Warning
+    ///
+    /// This method does not currently set the column widths. You will need to set them manually by
+    /// calling [`Table::widths`].
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
     ///
     /// # Examples
     ///
     /// ```rust
     /// # use ratatui::{prelude::*, widgets::*};
-    /// let table = Table::new(
-    ///     vec![
-    ///         Row::new(vec![
-    ///             Cell::from("Cell1"),
-    ///             Cell::from("Cell2")
-    ///         ]),
-    ///         Row::new(vec![
-    ///             Cell::from("Cell3"),
-    ///             Cell::from("Cell4")
-    ///         ]),
-    ///     ],
-    ///     [Constraint::Length(5), Constraint::Length(5)]
-    /// )
-    /// .block(Block::default().title("Table"));
+    /// let rows = [
+    ///     Row::new(vec!["Cell1", "Cell2"]),
+    ///     Row::new(vec!["Cell3", "Cell4"]),
+    /// ];
+    /// let table = Table::default().rows(rows);
     /// ```
-    pub fn block(mut self, block: Block<'a>) -> Self {
-        self.block = Some(block);
+    #[must_use = "fluent setter methods do not modify the underlying object"]
```

Comment:
Should probably change the `must_use` message to "method moves the value of self and returns the modified value" to stay consistent.

Reply:
done